phases/codesign.go

```diff
@@ -2,133 +2,295 @@ package phases
 
 import (
 	"fmt"
-	"io"
-	"io/ioutil"
-	"net/http"
-	"os"
 	"path/filepath"
-	"strings"
+	"sort"
 
-	"github.com/bitrise-io/go-utils/command"
+	bitriseModels "github.com/bitrise-io/bitrise/models"
+	"github.com/bitrise-io/codesigndoc/codesigndoc"
+	"github.com/bitrise-io/codesigndoc/models"
+	"github.com/bitrise-io/codesigndoc/xcode"
+	envmanModels "github.com/bitrise-io/envman/models"
+	"github.com/bitrise-io/go-utils/colorstring"
 	"github.com/bitrise-io/go-utils/log"
+	"github.com/bitrise-io/go-utils/pathutil"
+	"github.com/bitrise-io/go-utils/sliceutil"
+	"github.com/bitrise-io/goinp/goinp"
+	"github.com/bitrise-io/xcode-project/xcodeproj"
+	"github.com/bitrise-io/xcode-project/xcscheme"
+	"github.com/bitrise-io/xcode-project/xcworkspace"
 )
 
+// CodesignResultsIOS ...
+type CodesignResultsIOS struct {
+	certificates         models.Certificates
+	provisioningProfiles []models.ProvisioningProfile
+}
+
+// CodesignResultAndroid ...
+type CodesignResultAndroid struct {
+	KeystorePath, Password, Alias, KeyPassword string
+}
+
 // CodesignResult ...
 type CodesignResult struct {
-	KeystorePath, Password, Alias, KeyPassword string
-	ProfilePaths, CertificatePaths             []string
+	Android CodesignResultAndroid
+	IOS     CodesignResultsIOS
 }
 
-const (
-	platformIOS        = "iOS"
-	platformAndroid    = "Android"
-	platformBoth       = "iOS and Android"
-	codesignExportsDir = "codesigndoc_exports"
-)
+// Project types "web", "macos" are not suppored as do not have ios and android native projects
 
-func getPlatform(projectType string) string {
-	switch projectType {
-	case "ios":
-		return platformIOS
-	case "android":
-		return platformAndroid
-	case "xamarin", "flutter", "cordova", "ionic", "react-native":
-		return platformBoth
-	case "web", "macos":
-		// not supported
-	}
-	// other or if no project type specified
-	return ""
+// Ignoring for iOS, due to ucertainty that native Xcode project is present (as it is needed for archiving)
```

Comment:
same is true for Android, it is not guaranteed that the native Android project exists, ask for android keystore upload only in case of Android or other project type. For now pls go with this approach.